src/main/kotlin/flag/FlagConfigUpdater.kt

```diff
@@ -0,0 +1,271 @@
+package com.amplitude.experiment.flag
+
+import com.amplitude.experiment.LocalEvaluationConfig
+import com.amplitude.experiment.LocalEvaluationMetrics
+import com.amplitude.experiment.cohort.CohortLoader
+import com.amplitude.experiment.cohort.CohortStorage
+import com.amplitude.experiment.evaluation.EvaluationFlag
+import com.amplitude.experiment.util.*
+import com.amplitude.experiment.util.Logger
+import com.amplitude.experiment.util.daemonFactory
+import com.amplitude.experiment.util.wrapMetrics
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.concurrent.withLock
+import kotlin.math.max
+import kotlin.math.min
+
+internal interface FlagConfigUpdater {
+    // Start the updater. There can be multiple calls.
+    // If start fails, it should throw exception. The caller should handle error.
+    // If some other error happened while updating (already started successfully), it should call onError.
+    fun start(onError: (() -> Unit)? = null)
+    // Stop should stop updater temporarily. There may be another start in the future.
+    // To stop completely, with intention to never start again, use shutdown() instead.
+    fun stop()
+    // Destroy should stop the updater forever in preparation for server shutdown.
+    fun shutdown()
+}
+
+internal abstract class FlagConfigUpdaterBase(
```

Comment:
Add a quick class comment. What does this class do, what do extending classes modify and why.